plugins/utils/improved-search.nix

```
@@ -0,0 +1,122 @@
+{
+  lib,
+  helpers,
+  config,
+  pkgs,
+  ...
+}:
+with lib;
+# This plugin is only configured through keymaps, so we use `mkVimPlugin` without the
+# `globalPrefix` argument to avoid the creation of the `settings` option.
+  helpers.vim-plugin.mkVimPlugin config {
+    name = "improved-search";
+    originalName = "improved-search.nvim";
+    defaultPackage = pkgs.vimPlugins.improved-search-nvim;
+
+    maintainers = [maintainers.GaetanLepage];
+
+    extraOptions = {
+      keymaps = mkOption {
+        description = ''
+          Keymap definitions for search functions
+
+          See [here](https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators) for the list of available callbacks.
+        '';
+        type = with helpers.nixvimTypes;
+          listOf (
+            submodule {
+              options = {
+                key = mkOption {
+                  type = str;
+                  description = "The key to map.";
+                  example = "!";
+                };
+
+                mode = helpers.keymaps.mkModeOption "";
+
+                action = mkOption {
+                  type = with helpers.nixvimTypes;
+                    maybeRaw (
+                      # https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators
+                      enum [
+                        "stable_next"
+                        "stable_previous"
+                        "current_word"
+                        "current_word_strict"
+                        "in_place"
+                        "in_place_strict"
+                        "forward"
+                        "forward_strict"
+                        "backward"
+                        "backward_strict"
+                      ]
+                    );
+                  description = ''
+                    The action to execute.
+
+                    See https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators
```

IMO this'd be better as a markdown link, matching the keymaps description
```suggestion
                    See [here](https://github.com/backdround/improved-search.nvim?tab=readme-ov-file#functions-and-operators) for the list of available callbacks.
```

May also be nice to define the enum list in a `let` block and include it directly in the description text too?

Is it worth mentioning that the action can be raw lua, or does the type information make that obvious?

---

> May also be nice to define the enum list in a `let` block and include it directly in the description text too?

Not sure this would be necessary as the list of allowed values is already shown on the documentation.

> Is it worth mentioning that the action can be raw lua, or does the type information make that obvious?

Same, it will be visible on the documentation + Here it would be a fairly niche usage. So I would say no, we don't need to add it explicitly to the description.